keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,593 @@
+profiles at this point. User-defined profiles are essential for users to realize
+the full benefits out of AppArmor, allowing them to decrease their attack
+surface based on their own workloads.
+
+###### Updating localhost AppArmor profiles
+
+AppArmor profiles are applied at container creation time. The underlying
+container runtime only references already loaded profiles by its name.
+Therefore, updating the profiles content requires a manual reload (typically via
+`apparmor_parser`).
+
+Note that changing profiles is not recommended and may cause containers to fail
+on next restart, in the case of the new profile being more restrictive, invalid
+or the file no longer available on the host.
+
+Currently, users have no way to tell whether their physical profiles have been
+deleted or modified. This KEP proposes no changes to the existing functionality.
+
+The recommended approach for rolling out changes to AppArmor profiles is to
+always create _new profiles_ instead of updating existing ones. Create and
+deploy a new version of the existing Pod Template, changing the profile name to
+the newly created profile. Redeploy, once working delete the former Pod
+Template. This will avoid disruption on in-flight workloads.
+
+The current behavior lacks features to facilitate the maintenance of AppArmor
+profiles across the cluster. Two examples being: 1) the lack of profile
+synchronization across nodes and 2) how difficult it can be to identify that
+profiles have been changed on disk/memory, after pods started using it. However,
+Kubernetes managed profiles are out of scope for this KEP.
+Out of tree enhancements like the
+[security-profiles-operator](https://github.com/kubernetes-sigs/security-profiles-operator) can
+provide such enhanced functionality on top.
+
+###### Profiles managed by the cluster admins
+
+The current support relies on profiles being loaded on all cluster nodes
+where the pods using them may be scheduled. It is also the cluster admin's
+responsibility to ensure the profiles are correctly saved and synchronized
+across the all nodes. Existing mechanisms like node `labels` and `nodeSelectors`
+can be used to ensure that pods are scheduled on nodes supporting their desired
+profiles.
+
+##### RuntimeDefault Profile
+
+We propose maintaining the support to a single runtime profile, which will be
+defined by using the `AppArmorProfileTypeRuntimeDefault`. The reasons being:
+
+- No changes to the current behavior. Users are currently not allowed to specify
+  other runtime profiles. The existing API server rejects runtime profile names
+  that are different than `runtime/default`.
+- Most runtimes only support the default profile, although the CRI is flexible
+  enough to allow the kubelet to send other profile names.
+- Multiple runtime profiles has never been requested as a feature.
+
+If built-in support for multiple runtime profiles is needed in the future, a new
+KEP will be created to cover its details.
+
+#### Validation
```

Comment:
the feature is not clearly explaining what it means that AppArmor profile will be applied. The check for AppArmor being enabled or not is quite simple and not account for AppArmor service to be running. I actually don't know what will happen if one disabled the service. Will profile will still take an effect? I wonder what guarantees K8s node can give on whether AppArmor is working or not.

Reply:
I'm not sure exactly what you mean by this? The `apparmor.IsEnabled()` check checks whether the kernel module is loaded. We used to try and do this with the pod status, but I think if a user really wants to guarantee that apparmor is being enforced, they're better off using a 3rd party monitoring service that will actually confirm the process status (e.g. via `/proc/$PID/attr/current/`)

Reply:
It may be a good idea to add this note. So there is no expectations that k8s will somehow guarantee that AppArmor was not turned off and still being applied